demo/admin/src/products/ProductForm.tsx

```diff
@@ -84,6 +94,12 @@ function ProductForm({ id }: FormProps): React.ReactElement {
               ...filter<GQLProductFormManualFragment>(productFormFragment, data.product),
               price: String(data.product.price),
               image: rootBlocks.image.input2State(data.product.image),
+              packageDimensions: {
+                  ...data.product.packageDimensions,
+                  width: String(data.product.packageDimensions.width),
+                  height: String(data.product.packageDimensions.height),
+                  depth: String(data.product.packageDimensions.depth),
+              },
```

Comment:
Instead of stringifying/parsing we could also parse on field level, like we did for [SpaceBlock](https://github.com/vivid-planet/comet/blob/next/packages/admin/blocks-admin/src/blocks/SpaceBlock.tsx#L56). I don't know what I prefer though 🤷🏼‍♂️

Reply:
I will test that out. Sounds good, even the type modification could be avoided with that.

Reply:
See also Mr. final-form himself response here: https://stackoverflow.com/a/55121050